Fix defining wxNODISCARD for gcc

vadz opened the pull request:
Fix the check in f1985c6ba2 (Use [[nodiscard]] with functions returning object to be freed, 2022-11-04) for gcc which didn't work properly due to a typo.

See #22943.

vadz committed:
Fix defining wxNODISCARD for gcc

Fix the check in f1985c6ba2 (Use [[nodiscard]] with functions returning
object to be freed, 2022-11-04) for gcc which didn't work properly due
to a typo.

See #22943.

lanurmi commented:
The same typo in the same context is also present in `src/stc/lexilla/include/LexillaCompat.h`.

vadz committed:
Update Lexilla submodule to fix wxNODISCARD in it too

Make the same change as was done in the parent commit in the lexilla
submodule, which had the same typo.
